File: Data Structures & Algorithms/remove-element/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} val
     * @return {number}
     */
    removeElement(nums, val) {
        let start = 0, end = nums.length-1;
        let res = 0;
        while(start <= end){
            if(nums[start] === val){
                nums[start] = nums[end];
                end--;
            }
            else{
                res++;
                start++;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/house-robber/submission-1.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        # Use a dictionary to store the results of previously computed subproblems
        memo = {}
        
        def rob_house(i):
            # Base case: No more houses to rob
            if i >= len(nums):
                return 0
            
            # Check if the value for this house index is already calculated
            if i in memo:
                return memo[i]
            
            # Either rob this house and move to i + 2, or skip this house and move to i + 1
            memo[i] = max(nums[i] + rob_house(i + 2), rob_house(i + 1))
            
            return memo[i]
        
        # Start at the first house
        return rob_house(0)
        